src/KustoExpressionParser.test.ts

```diff
@@ -524,10 +524,17 @@ describe('KustoExpressionParser', () => {
           CslType: 'int',
           isDynamic: true,
         },
+        {
+          Name: 'column["level"]["active"]',
+          CslType: 'double',
+          isDynamic: true,
+        },
```

Comment:
I believe this is not represent reality. The schema will contain only one element per `Name` so, there won't be a repeated `column["level"]["active"]`.

The fix  should be in datasource.ts:

```ts
const recordSchemaArray = (name: string, types: AdxSchemaDefinition[], result: AdxColumnSchema[]) => {
  // If a field can have different types (e.g. long and double)
  // we select the first, assuming they are interchangeable
  const defaultCslType = types[0];
  if (types.every((t) => typeof t === 'string' && toPropertyType(t) === QueryEditorPropertyType.Number)) {
    // If all the types are numbers, the double takes precedence since it has more precission.
    const cslType = types.find((t) => typeof t === 'string' && (t === 'double' || t === 'real')) || defaultCslType;
    result.push({ Name: name, CslType: cslType as string, isDynamic: true });
  } else {
    console.warn(`schema ${name} may contain different types, assuming ${defaultCslType}`);
    if (typeof defaultCslType === 'object') {
      recordSchema(name, types[0], result);
    } else {
      result.push({ Name: name, CslType: defaultCslType, isDynamic: true });
    }
  }
};
```

here is where we record the schema in case the type has multiple values. We just need to remove the condition that checks if all the types are `Number` and simply set the type to double or real if it's part of the list.

Reply:
I see, so when we get to `escapeAndCastIfDynamic` the selection of a single type has already been made, my bad!

I'll make the changes to `datasource.ts` 🙇‍♀️